packages/auth/src/core/credentials/email.ts

```diff
@@ -123,32 +123,12 @@ export class EmailAuthCredential extends AuthCredential {
           password: this._password,
           clientType: RecaptchaClientType.WEB
         };
-        if (auth._getRecaptchaConfig()?.emailPasswordEnabled) {
-          const requestWithRecaptcha = await injectRecaptchaFields(
-            auth,
-            request,
-            RecaptchaActionName.SIGN_IN_WITH_PASSWORD
-          );
-          return signInWithPassword(auth, requestWithRecaptcha);
-        } else {
-          return signInWithPassword(auth, request).catch(async error => {
-            if (
-              error.code === `auth/${AuthErrorCode.MISSING_RECAPTCHA_TOKEN}`
-            ) {
-              console.log(
-                'Sign-in with email address and password is protected by reCAPTCHA for this project. Automatically triggering the reCAPTCHA flow and restarting the sign-in flow.'
-              );
-              const requestWithRecaptcha = await injectRecaptchaFields(
-                auth,
-                request,
-                RecaptchaActionName.SIGN_IN_WITH_PASSWORD
-              );
-              return signInWithPassword(auth, requestWithRecaptcha);
-            } else {
-              return Promise.reject(error);
-            }
-          });
-        }
+        return handleRecaptchaFlow(
```

Comment:
Instead of calling signInWithCredential in packages/auth/src/core/strategies/email_and_password.ts, can we instead duplicate a bit of code and do something like:

```
export function signInWithEmailAndPassword(
  auth: Auth,
  email: string,
  password: string
): Promise<UserCredential> {

        const request: SignInWithPasswordRequest = {
          returnSecureToken: true,
          email: this._email,
          password: this._password,
          clientType: RecaptchaClientType.WEB
        };
 const response = handleRecaptchaFlow(auth, request, action, signInWithPassword)
 .catch(error => {
    if (error.code === `auth/${AuthErrorCode.MFA_REQUIRED}`) {
      throw MultiFactorError._fromErrorAndOperation(
        auth,
        error,
        operationType,
        user
      );
    } else if (
      error.code === `auth/${AuthErrorCode.PASSWORD_DOES_NOT_MEET_REQUIREMENTS}`
    ) {
      void recachePasswordPolicy(auth);
    } else {
      throw error;
    }
  })
;

  const userCredential = await UserCredentialImpl._fromIdTokenResponse(
    auth,
    OperationType.SIGN_IN,
    response
  );

    await auth._updateCurrentUser(userCredential.user);
```

This will make the email login flow more consistent with the other flows in strategies/email_and_password.ts and we don't have to import any recaptcha enterprise methods here (enabling tree shaking out the JS script)

Reply:
Adding the comment here, but it is for the implementation in packages/auth/src/core/strategies/email_and_password.ts

Reply:
But doing this will not perform recaptcha enterprise wrapping if the developer is calling reauthenticateWithCredential or signInWithCredential directly... hmm.

Reply:
Yeah, I actually tried similar things when implementing the recapthca flow originally. I agree this part needs some refactors.